### DISCUSSION TITLE
Create a highlight group from a query

### DISCUSSION THREAD

**Discussion by toddteller** [nvim-treesitter/nvim-treesitter#4061] [Q&A] [answered]
How can I create a highlight group from this query?
```
(variable_declarator 
	(identifier) @id
		(call_expression
		 function: (identifier) @function (#eq? @function "require") 
	)   
)
```

**Comment by clason** [comment, marked as answer]
`:hi @function <whatever>`

(Highlight groups only exist for captures, not for composite queries.)

**Reply by toddteller** [reply]
I want to highlight all the occurrences of the variable in this code, any suggestions?
``` 
const variable = require (")
```

**Reply by theHamsta** [reply]
Can't you `:hi @id <something>`? Since it is a very special query you could give it a more specific name
```
(variable_declarator 
	(identifier) @required-variable
		(call_expression
		 function: (identifier) @_require (#eq? @_require "require") 
	)   
)
```
and give `highlight @required-variable <something>` some attributes (no idea whether vim script is ok with the `-`). We typically use underscore captures when we don't want to highlight something directly.

**Reply by toddteller** [reply]
So I have to create the highlight group with :hi and then execute the query? Sorry I am a beginner..

**Reply by theHamsta** [reply]
Just have the query anywhere in your runtime path (in a `<language here>/queries/highlights.scm`). You can use `:highlight @<capture name here> guibg=#000055` to set the background for a capture or specify any other details for that highlight or you can `:highlight @< capture name here> link Comment` to link your capture to any known highlight group (here `Comment`)

**Reply by theHamsta** [reply]
You can read more with `:h :highlight` or `:h :highlight-link`

**Reply by toddteller** [reply]
Thank you so much! I've solved
